doc/src/read_dump.rst

```diff
@@ -230,13 +230,18 @@ will then have a label corresponding to the fix-ID rather than "x" or
 "xs".  The *label* keyword can also be used to specify new column
 labels for fields *id* and *type*\ .
 
-For dump files in *xyz* format, only the *x*, *y*, and *z* fields are
-supported.  The dump file does not store atom IDs, so these are
-assigned consecutively to the atoms as they appear in the dump file,
-starting from 1.  Thus you should ensure that order of atoms is
-consistent from snapshot to snapshot in the XYZ dump file.  See
-the :doc:`dump_modify sort <dump_modify>` command if the XYZ dump file
-was written by LAMMPS.
+For dump files in *xyz* format, only the *type*, *x*, *y*, and *z*
+fields are supported.  If the atoms have an integer number as label
+(like when they are created with default setting by :doc:`dump style
+<dump>` *xyz*), that number will be used at atom type. The the atom
+label is a string, it is required that a map is created using the
+:doc:`labelmap command <labelmap>` that maps those labels to the
+(numeric) atom types that LAMMPS uses.  The xyz format dump file does
+not store atom IDs, so these are assigned consecutively to the atoms as
+they appear in the dump file, starting from 1.  Thus you should ensure
+that order of atoms is consistent from snapshot to snapshot in the XYZ
+dump file.  See the :doc:`dump_modify sort <dump_modify>` command if the
+XYZ dump file was written by LAMMPS.
```

Comment:
I didn't fully understand what was meant by the new text, but if I understood correctly, how about:

If the label provided for atoms is an integer number 
(like when they are created with default setting by :doc:`dump style
<dump>` *xyz*), that number will be used as the atom type. If the atom
label is a string, it is required that a map is created using the
:doc:`labelmap command <labelmap>` that maps those string labels to the
(numeric) atom types that LAMMPS uses.  The xyz format dump file does
not store atom IDs, so these are assigned consecutively to the atoms as
they appear in the dump file, starting from 1.  Thus you should ensure
that the order of atoms is consistent from snapshot to snapshot in the XYZ
dump file.  See the :doc:`dump_modify sort <dump_modify>` command if the
XYZ dump file was written by LAMMPS.

(also fixed a few minor typos)

Reply:
Previously, the label would be converted by atoi(), which does not flag an error with incorrect or incomplete conversions and would either return any leading numbers or 0. Now we are always returning a valid type. Either the number we have read (and checked to be valid) or the number from the labelmap.

I have reformulated the text a little bit for more clarity.

@sjplimp the description says that the typemap is required for "string labels" it said nothing about numeric labels. I've added a remark that those remain unchanged.

Reply:
@sjplimp @jtclemm is this now sufficiently clear in the pull request? If not, please let me know where there are still problem. thanks.